app/tests/sitemap.test.js

```diff
@@ -0,0 +1,38 @@
+// Test ../routes/sitemap.js
+
+import fs from 'fs'
+import request from 'supertest'
+import express from 'express'
+import sitemapRouter from '../routes/sitemap'
+
+jest.mock('fs')
+jest.mock('../tools/getSitemap', () => jest.fn())
+
+const app = express()
+app.use('/', sitemapRouter)
+
+describe('GET /sitemap.xml', () => {
+  it('should respond with sitemap.xml content', async () => {
+    const sitemapContent = '<urlset><url><loc>http://example.com/</loc></url></urlset>'
+    fs.readFile.mockImplementation((filePath, callback) => {
```

Comment:
Starting up a server to test the routes sounds like a great idea.

After you implment all the code in the route, You will need to startup mongmemory server as in app/socket-apis/__tests__/get-random-whys.js see beforeAll and afterAll

for what to expect from the test, you can just expect it to match a string embeded in this file. Or iotas.json will change from time to time, causing the sitemap to change. but mostly it will be new things getting added. Maybe you can just expect the returned sitemap to contain certain things. 

In the file you will need 
import Iotas from "civil-server" so that the Iotas model will get loaded, and this will load iotas.json into an empty database.